oci/casext/gc_test.go

```diff
@@ -175,3 +176,144 @@ func TestGCWithNonEmptyIndex(t *testing.T) {
 		t.Fatalf("expected single-entry blob list after GC")
 	}
 }
+
+func gcOKFunc(desc ispec.Descriptor) (bool, error) {
+	return true, nil
+}
+
+func gcSkipFunc(desc ispec.Descriptor) (bool, error) {
+	return false, nil
+}
+
+func errFunc(desc ispec.Descriptor) (bool, error) {
+	return false, fmt.Errorf("err policy")
+}
```

Comment:
Ideally these should be functions that actually check that the `Digest` passed is correct. A simple way to do this would be:

```go
func gcOkFunc(t *testing.T, expectedDigest digest.Digest) GCPolicy {
	return func(digest digest.Digest) (bool, error) {
		if digest != expectedDigest {
			t.Errorf("got incorrect digest to gc policy callback: expected %v, got %v", expectedDigest, digest)
		}
		return true, nil
	}
}
```